test/threads_exec.jl

```diff
@@ -840,6 +840,18 @@ fib34666(x) =
     end
 @test fib34666(25) == 75025
 
+# issue #41324
+@testset "Co-schedule" begin
+    parent = Threads.@spawn begin
+        @test current_task().sticky == false
+        child = @async begin end
+        @test current_task().sticky == true
```

Comment:
Seems like it would be better to compare `threadid()` to check that they're the same.